Update appearance of setting default value indicator

bdach opened the pull request:
Loosely based on the following designs: https://www.figma.com/file/IjrU0u7MFdgmxDofqTlsZb/UI%2FClient-Settings?node-id=0%3A1

https://user-images.githubusercontent.com/20418176/137644785-fdc5df69-49ea-46b3-aef7-8eb0e70beddd.mp4

The idle/default behaviour was provided in the design, but there was no hover/disabled behaviour, so I improvised. The clickable area is quite a bit larger than what it looks to be to alleviate pixel hunting, although I'm not super sure if it's large enough still.

The dot will appear next to the setting control if the setting has no label, and next to the label if there is one. This was the most finicky part of this implementation by far. It looks to work pretty well, though.

Colours taken from https://www.figma.com/file/VIkXMYNPMtQem2RJg9k2iQ/Asset%2FColours?node-id=0%3A1.

bdach committed:
Restyle default value indicator

bdach committed:
Adjust `SettingsItem` to accommodate new default value indicator

bdach committed:
Adjust key binding rows to accommodate new default value indicator

bdach committed:
Remove unneeded extra padding from settings number box

smoogipoo commented:
Imo this is going to be hard to hit on mobile but I guess we're not considering those platforms yet.

peppy commented:
It is a consideration, but it's not much worse than the previous version. Mobile should have a higher default UI scale to make these kinds of things larger.

peppy committed:
Refactor test to better keep existing toggle values

I also changed the type of the button to `float` because it was mentally
hard to parse a default button that is tracking a `bool` state. Probably
not what we want for a test like this.

bdach committed:
Remove doubled-up opacity specification

bdach committed:
Remove grid usage in `KeyBindingRow`

bdach committed:
Remove grid usage in `SettingsItem`

bdach committed:
Merge branch 'master' into default-value-indicator

bdach committed:
Fix settings item having zero height

peppy committed:
Merge branch 'master' into default-value-indicator

peppy commented:
Hmmm I see. Kind of a weird way of laying things out, but let's see how users find it.

I will say that when it's to the left of the label/header element, I'd expect to be able to click anywhere on the label to trigger the reset-to-default. Quite a few photos/graphics editor apps do this, so it's a pretty proven UX.
